packages/analytics-core/src/plugins/destination.ts

```diff
@@ -269,6 +297,11 @@ export class Destination implements DestinationPlugin {
       return true;
     });
 
+    const dropEvents = list.filter((element) => !retry.includes(element));
```

Comment:
Efficiency - the same list is already filtered above which could be heavy for large lists. Also `list.includes()` is linear. `Set.has()` is O(1) and can be better for frequent lookups.

It would probably be more efficient to do it all at once. e.g.
```
const retryList = [];
const dropList = [];
list.forEach(element => {
   if(shouldRetry(element)) {
     ...
     retryList.push();
   } else {
      ...
      dropList.push();
   }
})
```

Reply:
How about keeping the original code and only add `dropList.push()` 

```
const dropList = []

    const retry = list.filter((context, index) => {
      if ( shouldDrop ) {
        this.fulfillRequest([context], res.statusCode, res.body.error);
        dropList.push(context);
        return;
      }
      ...
      return true;
    });
```
It also only iterates once.

Reply:
That sounds good.

Reply:
Done in https://github.com/amplitude/Amplitude-TypeScript/pull/568/commits/025b4b079a1a2f452918647f99897abff920f476